controller/konnect/ops/ops_controlplane_test.go

```diff
@@ -448,3 +457,62 @@ func TestUpdateControlPlane(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateAndUpdateControlPlane_KubernetesMetadataConsistency(t *testing.T) {
+	var (
+		ctx = context.Background()
+		cp  = &konnectv1alpha1.KonnectGatewayControlPlane{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "KonnectGatewayControlPlane",
+				APIVersion: "konnect.konghq.com/v1alpha1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "cp-1",
+				Namespace:  "default",
+				UID:        k8stypes.UID(uuid.NewString()),
+				Generation: 2,
+			},
+			Spec: konnectv1alpha1.KonnectGatewayControlPlaneSpec{
+				CreateControlPlaneRequest: sdkkonnectcomp.CreateControlPlaneRequest{
+					Name: "cp-1",
+				},
+			},
+		}
+		sdk = &MockControlPlaneSDK{}
```

Comment:
Please use the mocks constructor which ensures that `AssertExpectations` is called. An alternative is to `defer`/`t.Cleanup(...)` the `AssertExpectations` call but the constructor approach is better IMHO.

For reference: https://github.com/Kong/gateway-operator/pull/574